libcontainer/process_linux.go

```diff
@@ -439,6 +483,47 @@ func (p *initProcess) start() (retErr error) {
 
 	ierr := parseSync(p.messageSockPair.parent, func(sync *syncT) error {
 		switch sync.Type {
+		case procSeccomp:
+			// receive seccomp-fd
+			pidfd, _, errno := unix.Syscall(unix.SYS_PIDFD_OPEN, uintptr(childPid), 0, 0)
+			if errno != 0 {
+				return newSystemErrorWithCause(errno, "performing SYS_PIDFD_OPEN syscall")
+			}
+			defer unix.Close(int(pidfd))
+
+			seccompFd, _, errno := unix.Syscall(unix.SYS_PIDFD_GETFD, pidfd, uintptr(sync.Fd), 0)
+			if errno != 0 {
+				return newSystemErrorWithCause(errno, "performing SYS_PIDFD_GETFD syscall")
+			}
+			defer unix.Close(int(seccompFd))
+
+			if p.config.Config.Seccomp.ListenerPath == "" {
+				return newSystemError(errors.New("listenerPath is not set"))
+			}
```

Comment:
Perhaps it makes sense to consolidate this code since it's repeated twice. Something like

```go
seccompFd, err := recvSeccompFd(childPid, sync.Fd)
```

maybe?

Reply:
Yeap, good call. I'm moving only these bits and not the `if p.config.Config.Seccomp.ListenerPath == "" ` part as we need `p` for that and it would be called with `p` being different structs. Also, IMHO, semantically seems better to keep both things separate